dlrover/trainer/tensorflow/util/dataset_util.py

```diff
@@ -43,15 +41,18 @@ def __init__(
         self._epoch = epoch
         scheme, path = path_util.parse_uri(path)
         self.path = path
-        if scheme == TFConstants.FILE_SCHEME():
-            self.reader = FileReader(
-                self.path, batch_size=self._batch_size, num_epochs=self._epoch
-            )
-        elif scheme == TFConstants.FAKE_SCHEME():
-            self.reader = FakeReader()  # type: ignore
+        self.initialize_reader(scheme, path)
 
-        if reader_fn is not None:
-            self.reader = reader_fn
+    def initialize_reader(self, scheme, path):
+        reader_type = scheme.split(":")[0]
+        print(reader_type)
+        Reader = reader_registery.get_reader(reader_name=reader_type)
```

Comment:
Here, users cannot define different readers for different datasets.  It is better to `get_reader` by the dataset name.